storage.go

```diff
@@ -164,3 +243,38 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
 
 	return size, nil
 }
+
+func (s *Storage) writeAppend(ctx context.Context, o *Object, r io.Reader, size int64, opt pairStorageWriteAppend) (n int64, err error) {
+	if !o.Mode.IsAppend() {
+		err = fmt.Errorf("object not appendable")
+		return
+	}
+
+	sessionId := GetObjectMetadata(o).UploadSessionID
+
+	offset, ok := o.GetAppendOffset()
```

Comment:
Maybe we can make sure that `append-offset` is set while object has mode `append`